js/src/containers/DocumentSidebar.js

```diff
@@ -1,39 +1,61 @@
 import { withSelect, withDispatch } from "@wordpress/data";
 import { compose } from "@wordpress/compose";
 import getIndicatorForScore from "../analysis/getIndicatorForScore";
-import DocumentSidebar from "../components/DocumentSidebar";
+import AnalysisChecklist from "../components/AnalysisChecklist";
+
+import {
+	addReadabilityCheck,
+	addSEOCheck,
+	addSchemaBlocksValidationCheck,
+} from "../helpers/addCheckToChecklist";
+
+/**
+ * Maps the select function to props for the checklist.
+ *
+ * @param {function} select The WordPress select function.
+ *
+ * @returns {{shouldShowIntro: boolean, checklist: []}} The props for the checklist.
+ */
+export function mapSelectToProps( select ) {
+	const data = select( "yoast-seo/editor" );
+	const seoScoreIndicator = getIndicatorForScore( data.getResultsForFocusKeyword().overallScore );
+	const readabilityScoreIndicator = getIndicatorForScore( data.getReadabilityResults().overallScore );
+	const { isKeywordAnalysisActive, isContentAnalysisActive } = data.getPreferences();
+	const schemaBlocksValidationResults = data.getSchemaBlocksValidationResults();
+
+	const checklist = [];
+
+	addReadabilityCheck( checklist, readabilityScoreIndicator, isKeywordAnalysisActive );
```

Comment:
Instead of passing `isKeywordAnalysisActive` I would just do the lookup in the function itself.